pkg/controllers/monitoring/monitoring-stack/components.go

```diff
@@ -23,62 +23,63 @@ const PrometheusUserFSGroupID = 65534
 const AlertmanagerUserFSGroupID = 65535
 
 func stackComponentReconcilers(ms *stack.MonitoringStack, instanceSelectorKey string, instanceSelectorValue string) []reconciler.Reconciler {
-	prometheusRBACResourceName := ms.Name + "-prometheus"
-	alertmanagerRBACResourceName := ms.Name + "-alertmanager"
+	prometheusName := ms.Name + "-prometheus"
+	alertmanagerName := ms.Name + "-alertmanager"
 	rbacVerbs := []string{"get", "list", "watch"}
 	additionalScrapeConfigsSecretName := ms.Name + "-prometheus-additional-scrape-configs"
+	hasNsSelector := ms.Spec.NamespaceSelector != nil
+
 	return []reconciler.Reconciler{
-		reconciler.NewUpdater(newServiceAccount(prometheusRBACResourceName, ms.Namespace), ms),
-		reconciler.NewUpdater(newPrometheusRole(ms, prometheusRBACResourceName, rbacVerbs), ms),
-		reconciler.NewUpdater(newRoleBinding(ms, prometheusRBACResourceName), ms),
+		reconciler.NewUpdater(newServiceAccount(prometheusName, ms.Namespace), ms),
+		reconciler.NewUpdater(newPrometheusRole(ms, prometheusName, rbacVerbs), ms),
+
+		// create clusterrole if there is a nsSelector otherwise delete any that was created earlier
```

Comment:
nit: The comment and what comes after do not match. Here we are talking about clusterrole and the line below talks about clusterrolebinding.

Reply:
Well spotted! it must be clusterrolebinding